vehicle/autoware_external_cmd_converter/tests/test_external_cmd_converter.cpp

```diff
@@ -0,0 +1,169 @@
+    }
+    const int sleep_time = static_cast<int>(external_cmd_converter_->control_command_timeout_) + 1;
+    std::this_thread::sleep_for(std::chrono::seconds(sleep_time));
+    return external_cmd_converter_->check_remote_topic_rate();
+  }
+
+  bool test_check_emergency_stop_topic_timeout(
+    bool received_data, bool is_auto, bool time_has_passed)
+  {
+    if (!received_data) {
+      return external_cmd_converter_->check_emergency_stop_topic_timeout();
+    }
+    tier4_external_api_msgs::msg::Heartbeat::ConstSharedPtr msg;
+    external_cmd_converter_->on_emergency_stop_heartbeat(msg);
+    GateMode gate;
+    gate.data = (is_auto) ? tier4_control_msgs::msg::GateMode::AUTO
+                          : tier4_control_msgs::msg::GateMode::EXTERNAL;
+    external_cmd_converter_->current_gate_mode_ = std::make_shared<GateMode>(gate);
+
+    const int sleep_time = static_cast<int>(external_cmd_converter_->emergency_stop_timeout_) + 1;
+    if (time_has_passed) std::this_thread::sleep_for(std::chrono::seconds(sleep_time));
+    return external_cmd_converter_->check_emergency_stop_topic_timeout();
+  }
+
+  double test_get_shift_velocity_sign(const GearCommand & cmd)
+  {
+    return external_cmd_converter_->get_shift_velocity_sign(cmd);
+  }
+
+  void initialize_maps(const std::string & accel_map, const std::string & brake_map)
+  {
+    external_cmd_converter_->accel_map_.readAccelMapFromCSV(accel_map);
+    external_cmd_converter_->brake_map_.readBrakeMapFromCSV(brake_map);
+    external_cmd_converter_->acc_map_initialized_ = true;
+  }
+
+  double test_calculate_acc(
+
+    const ExternalControlCommand & cmd, const double vel)
+  {
+    return external_cmd_converter_->calculate_acc(cmd, vel);
+  }
+
+  std::shared_ptr<ExternalCmdConverterNode> external_cmd_converter_;
+};
+
+TEST_F(TestExternalCmdConverter, testCheckEmergencyStopTimeout)
+{
+  EXPECT_TRUE(test_check_emergency_stop_topic_timeout(false, false, false));
+  EXPECT_FALSE(test_check_emergency_stop_topic_timeout(true, true, false));
+  EXPECT_TRUE(test_check_emergency_stop_topic_timeout(true, false, false));
+  EXPECT_FALSE(test_check_emergency_stop_topic_timeout(true, false, true));
+}
+
+TEST_F(TestExternalCmdConverter, testRemoteTopicRate)
+{
+  EXPECT_TRUE(test_check_remote_topic_rate(false, false, false));
+  EXPECT_TRUE(test_check_remote_topic_rate(true, true, false));
+  EXPECT_FALSE(test_check_remote_topic_rate(true, true, true));
+}
+
+TEST_F(TestExternalCmdConverter, testGetShiftVelocitySign)
+{
+  GearCommand cmd;
+  cmd.command = GearCommand::DRIVE;
+  EXPECT_DOUBLE_EQ(1.0, test_get_shift_velocity_sign(cmd));
+  cmd.command = GearCommand::LOW;
+  EXPECT_DOUBLE_EQ(1.0, test_get_shift_velocity_sign(cmd));
+  cmd.command = GearCommand::REVERSE;
+  EXPECT_DOUBLE_EQ(-1.0, test_get_shift_velocity_sign(cmd));
+  cmd.command = GearCommand::LOW_2;
+  EXPECT_DOUBLE_EQ(0.0, test_get_shift_velocity_sign(cmd));
+}
+
+TEST_F(TestExternalCmdConverter, testCalculateAcc)
```

Comment:
Could you make test cases for boundary conditions? For example, a case with `throttle = Inf (large number)`?

Reply:
@TakaHoribe thanks for your suggestions. I have added descriptions to the tests and functions used to evaluate. I also added combinations of std::numeric_limits<double>::infinity() to the throttle and brake. Could you please check again?

Reply:
You only check if the output is positive/negative for infinite cases. Are you sure that those outputs are not nan nor other invalid numbers?
This PR looks ok if the above concern is solved.